src/main/java/com/salesforce/dynamodbv2/mt/repo/MtDynamoDbTableDescriptionRepo.java

```diff
@@ -145,6 +156,11 @@ public TableDescription deleteTable(String tableName) {
         return tableDescription;
     }
 
+    private TenantTable getTenantTableFromHashKey(String hashKey) {
```

Comment:
Can we just use `TenantTable` as the map key? Using a string can cause collisions, so it's been on our todo list to replace it.

Reply:
not sure I follow... the hash key within the metadata table at the moment is a string (tenantName + delimeter + virtualTableName). The map key therefore is to take a row from Dynamo, and serialize the row with its `TenantTable` and `CreateTableRequest` onto S3 per tenant-table backup. 
Did you mean use `TenantTable` serialized into a binary hash key onto our dynamo metadata table?

Reply:
Sorry, I got mixed up. In memory MtCache adds the prefix and we already changed that from string concatenation to a data class. The hash key for the DynamoDB table is still computed with unsafe string concatenation; to fix that we’d have to use something similar to what we’re doing for data tables (String/BinaryPrefixFunction). That may require dropping the metadata table, so we should tackle that separately.